rbac_api_with_domains.go

```diff
@@ -60,55 +60,70 @@ func (e *Enforcer) DeleteRolesForUserInDomain(user string, domain string) (bool,
 }
 
 // GetAllUsersByDomain would get all users associated with the domain.
+// index denotes the index of domain in p section starting at 0.
 func (e *Enforcer) GetAllUsersByDomain(domain string) []string {
 	m := make(map[string]struct{})
 	g := e.model["g"]["g"]
-	if len(g.Tokens) != 3 {
-		return []string{}
-	}
+	p := e.model["p"]["p"]
 	users := make([]string, 0)
-	for _, policy := range g.Policy {
-		if _, ok := m[policy[2]]; policy[2] == domain && ok {
-			users = append(users, policy[0])
+	index := e.getDomainIndex("p")
+
+	getUser := func(index int, policies [][]string) {
+		if len(policies) == 0 || len(policies[0]) <= index {
```

Comment:
why not return users in func? I suggest removing all implicit args like domain, users